tools/clang/lib/CodeGen/CGExprCXX.cpp

```diff
@@ -217,6 +217,27 @@ RValue CodeGenFunction::EmitCXXMemberOrOperatorMemberCallExpr(
       assert(opCall->getOperator() == OverloadedOperatorKind::OO_Subscript &&
              "must be subscript");
 
+      auto castIfBool = [&](llvm::Value *This, llvm::Value *Val) {
+        bool isBool = false;
+        if (llvm::IntegerType *IT =
+                dyn_cast<llvm::IntegerType>(This->getType()
+                                                ->getPointerElementType()
+                                                ->getVectorElementType())) {
+          if (IT->getBitWidth() == 1) {
+            isBool = true;
+          }
+        }
+        if (isBool) {
+          // bool pointer is not i1 *.
+          llvm::Type *BoolTy = llvm::IntegerType::get(
+              getLLVMContext(), getContext().getTypeSize(CE->getType()));
+          Val = Builder.CreateBitCast(
+              Val, llvm::PointerType::get(
+                       BoolTy, Val->getType()->getPointerAddressSpace()));
+        }
+        return Val;
```

Comment:
nit:
```suggestion
        Type *ElTy = This->getType()->getPointerElementType()
                                    ->getVectorElementType();
        if (llvm::IntegerType *IT = dyn_cast<llvm::IntegerType>(ThisTy))
          if (IT->getBitWidth() != 1)
            return Val;

        // bool pointer is not i1 *.
        llvm::Type *BoolTy = llvm::IntegerType::get(
            getLLVMContext(), getContext().getTypeSize(CE->getType()));
        Val = Builder.CreateBitCast(
            Val, llvm::PointerType::get(
                     BoolTy, Val->getType()->getPointerAddressSpace()));
        return Val;
```

Reply:
My original patch actually looked like this, but I wanted to make it clear that I had simply refactored the existing code mostly as-is. Will apply this change.

Reply:
Uploaded new patch based on our conversation.